src/main/java/aeneas/models/Level.java

```diff
@@ -135,5 +153,38 @@ public static Level loadLevel(File file) throws IOException {
     return bullpen.pieces;
   }
 
-  public abstract LevelWidgetView makeCorrespondingView();
+  public abstract LevelWidgetView makeCorrespondingView(Model model);
+
+  public abstract RadioButton getButton();
+
+  public abstract String getIconName();
+
+  public void start() { active = true; }
+  public void stop() { active = false; }
+
+  public boolean isActive() {
+    return active;
+  }
+
+  /**
+   * Get the text for counting down the user to finished.
+   * @return The text to be displayed to the user, eg "Time Remaining: 50"
+   */
+  public abstract String getCountdownText();
```

Comment:
I think the name could be better, but all levels have some kind of text indicating number of moves or time remaining, and we're currently using the same field for both.